### DISCUSSION TITLE
Unable to build Cockpit locally

### DISCUSSION THREAD

**Discussion by Prince2319** [cockpit-project/cockpit#18408] [Q&A] [answered]
![Screenshot from 2023-02-26 21-31-13](https://user-images.githubusercontent.com/123302204/221421723-cdd80e19-763c-4396-966c-b2c9341771f0.png)

I have installed gettext tools then also this isssue is not resolving can anyone please help me

**Comment by martinpitt** [comment, marked as answer]
The configure check is indeed slightly wrong/misleading. I fixed that in https://github.com/cockpit-project/cockpit/pull/18409 .

Please install `/usr/bin/xsltproc` (the package name depends on your distro, but you can usually search for that); a later configure check requires that anyway, at least if you build with `--enable-docs`.

**Comment by Prince2319** [comment]
![Screenshot from 2023-02-27 13-09-25](https://user-images.githubusercontent.com/123302204/221502764-e9a53ca6-d264-4c11-94ce-ce7a82581b44.png)

Sir now this error is coming

**Reply by jelly** [reply]
Can you show the output of `git status`. My suspicion is that you ran `npm install` or `yarn install` which changes either the `package.json` or package-lock.json` file.

**Reply by martinpitt** [reply]
Right, please run `tools/node-modules checkout` to fix this. We don't want developers to run `npm install` directly.

**Comment by Prince2319** [comment]
![Screenshot from 2023-02-27 14-35-27](https://user-images.githubusercontent.com/123302204/221520313-4c7efcb6-02dd-4277-ae90-fd2171ddec68.png)

SIr this is showing

**Comment by Prince2319** [comment]
![Screenshot from 2023-02-27 14-42-41](https://user-images.githubusercontent.com/123302204/221521904-349fa5da-fdbc-4904-8f99-a68d43e81b3f.png)

Sir by running this command tools/node-modules checkout it is showing this error

**Reply by jelly** [reply]
You'll have to undo the changes to `node_modules` and `package.json` with:

```
git reset HEAD --hard
```

And then run `./tools/node-modules checkout`

**Comment by Prince2319** [comment]
![Screenshot from 2023-02-27 14-48-13](https://user-images.githubusercontent.com/123302204/221523193-8c9fb923-e1bd-41d5-a1f8-78a8a2a8d3d3.png)

Sir now can i run make
